### PR TITLE
Add unittest for utils_func

### PR DESCRIPTION
**Pull request by zhoonit**
Add some unittests and fix errors for utils
 - changed answer to be declaritive
 - fix tolerence is calculated one way (should be wrapped with fabs)
 - add prime test unit tests

Self evaluation:

 Build test: [X]Passed [ ]Failed [ ]Skipped
 Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

File: test/unittest/unittest_util_func.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@
       }                                      \
     }                                        \
   } while (0)
-
+
```
Comment (gichan-jang) [inline test/unittest/unittest_util_func.cpp:44]: trailing whitespace

File: test/unittest/unittest_util_func.cpp [hunk 2]
```diff
@@ -136,7 +172,7 @@ TEST(nntrainer_util_func, sigmoid_01_p) {
   ASSERT_NE(nullptr, indata);
 
   for (int i = 0; i < batch * height * width; ++i) {
-    if ((data[i] - (float)(1 / (1 + exp(-indata[i])))) > tolerance) {
+    if (fabs(data[i] - answer[i]) > tolerance) {
```
Comment (gichan-jang) [inline test/unittest/unittest_util_func.cpp:175]: how about using `EXPECT_NEAR(val1, val2, abs_error);`? 
I have never used it, but I think the code will be simpler.
reference link ; https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#floating-point-macros

File: test/unittest/unittest_util_func.cpp (hunk 1 above)
Comment (zhoonit) [reply test/unittest/unittest_util_func.cpp]: Thanks. Fixed!

File: test/unittest/unittest_util_func.cpp (hunk 2 above)
Comment (zhoonit) [reply test/unittest/unittest_util_func.cpp]: Thank you for the suggestion! I changed it :)

File: test/unittest/unittest_util_func.cpp [hunk 3]
```diff
@@ -88,13 +106,12 @@ TEST(nntrainer_util_func, sqrtFloat_01_p) {
   float x = 9871.0;
   float sx = nntrainer::sqrtFloat(x);
 
-  if ((sx * sx - x) * (sx * sx - x) < tolerance)
+  if (fabs(sx * sx - x) * (sx * sx - x) < tolerance)
```
Comment (jijoongmoon) [inline test/unittest/unittest_util_func.cpp:109]: just ``` fabs(sx*sx-x)``` is ok.

Comment (zhoonit) [reply test/unittest/unittest_util_func.cpp:109]: Fixed. Thanks!

**Pull request merged**